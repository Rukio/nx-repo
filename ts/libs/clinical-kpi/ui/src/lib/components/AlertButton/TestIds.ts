export const ALERT_BUTTON_TEST_IDS = {
  TEXT: 'alert-button-text',
  BUTTON: 'alert-button-button',
};
