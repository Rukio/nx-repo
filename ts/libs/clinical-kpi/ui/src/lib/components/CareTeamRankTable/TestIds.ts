export const CARE_TEAM_RANK_TABLE_TEST_IDS = {
  TABLE: 'care-team-rank-table',
  getRankTestId: (rank?: number) => `care-team-table-rank-${rank}`,
  getProviderNameTestId: (rank?: number) =>
    `care-team-rank-table-provider-name-${rank}`,
  getValueTestId: (rank?: number) => `care-team-rank-table-value-${rank}`,
  getValueChangeTestId: (rank?: number) =>
    `care-team-rank-table-value-change-${rank}`,
  getPlaceholderTestId: (rank?: number) =>
    `care-team-rank-table-placeholder-${rank}`,
};

export const CARE_TEAM_PLACEHOLDER_ROW_RANKS = [1, 2, 3, 4, 5, 6, 7];
