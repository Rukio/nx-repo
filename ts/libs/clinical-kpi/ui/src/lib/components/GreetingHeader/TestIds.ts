export const GREETING_HEADER_TEST_IDS = {
  GREETING_TEXT: 'header-greeting-text',
  LAST_UPDATED_TEXT: 'header-last-updated-text',
  VISITS_COMPLETED_TEXT: 'header-visits-completed-text',
  PERFORMANCE_TEXT: 'header-performance-text',
  LEARN_MORE_LINK: 'header-learn-more-link',
  LAST_UPDATED_PLACEHOLDER: 'header-last-updated-placeholder',
  VISITS_COMPLETED_PLACEHOLDER: 'header-visits-completed-placeholder',
  STATION_LINK: 'header-station-dashboard-link',
  STATION_LINK_LABEL_PREFIX: 'header-station-dashboard-link-label-prefix',
};
