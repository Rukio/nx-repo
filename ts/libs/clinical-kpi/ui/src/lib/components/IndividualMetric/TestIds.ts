export const INDIVIDUAL_METRIC_TEST_IDS = {
  TYPE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-type`,
  VALUE: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-value`,
  VALUE_CHANGE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-change`,
  PERFORMANCE: (testIdPrefix: string) =>
    `${testIdPrefix}-individual-metric-performance`,
  GOAL: (testIdPrefix: string) => `${testIdPrefix}-individual-metric-goal`,
};
