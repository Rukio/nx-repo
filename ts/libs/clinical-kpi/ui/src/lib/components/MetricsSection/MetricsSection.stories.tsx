import { PersonIcon } from '@*company-data-covered*/design-system';
import { StoryFn, Meta } from '@storybook/react';
import { MetricsSection } from './';

export default {
  component: MetricsSection,
  title: 'MetricsSection',
} as Meta<typeof MetricsSection>;

const Template: StoryFn<typeof MetricsSection> = (args) => (
  <MetricsSection {...args} />
);

export const Primary = Template.bind({});
Primary.args = { title: 'My Performance', icon: <PersonIcon /> };
