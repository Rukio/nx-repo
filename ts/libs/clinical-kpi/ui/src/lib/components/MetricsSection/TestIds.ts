export const METRICS_SECTION_TEST_IDS = {
  TITLE: 'metrics-section-title',
};
