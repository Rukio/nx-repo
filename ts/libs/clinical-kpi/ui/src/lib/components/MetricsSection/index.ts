export { default as MetricsSection } from './MetricsSection';
export { METRICS_SECTION_TEST_IDS } from './TestIds';
