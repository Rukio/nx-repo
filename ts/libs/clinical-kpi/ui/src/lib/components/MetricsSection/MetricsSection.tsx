import { FC, ReactNode } from 'react';
import {
  Container,
  Box,
  makeSxStyles,
  Typography,
  TypographyProps,
} from '@*company-data-covered*/design-system';
import { METRICS_SECTION_TEST_IDS } from './TestIds';

type Props = {
  testIdPrefix?: string;
  title: string;
  icon?: ReactNode;
  select?: ReactNode;
  titleColor?: TypographyProps['color'];
  variant?: TypographyProps['variant'];
  children: ReactNode;
};

const makeStyles = ({ isIconDisplayed }: { isIconDisplayed: boolean }) =>
  makeSxStyles({
    contentContainer: (theme) => ({
      display: 'flex',
      borderBottom: 2,
      borderBottomColor: 'divider',
      borderBottomStyle: 'solid',
      padding: `${theme.spacing(0.5)} 0px`,
      alignItems: 'center',
    }),
    titleWrapper: (theme) => ({
      marginLeft: isIconDisplayed ? theme.spacing(1.5) : 0,
    }),
  });

const MetricsSection: FC<Props> = ({
  title,
  icon,
  select,
  testIdPrefix,
  titleColor = 'primary',
  variant = 'h6',
  children,
}) => {
  const styles = makeStyles({ isIconDisplayed: Boolean(icon) });

  return (
    <Container>
      <Box sx={styles.contentContainer}>
        {icon}
        <Typography
          data-testid={`${testIdPrefix}-${METRICS_SECTION_TEST_IDS.TITLE}`}
          sx={styles.titleWrapper}
          variant={variant}
          color={titleColor}
        >
          {title}
        </Typography>
        {select}
      </Box>
      {children}
    </Container>
  );
};

export default MetricsSection;
