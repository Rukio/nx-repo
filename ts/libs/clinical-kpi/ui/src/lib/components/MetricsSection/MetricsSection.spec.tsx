import { PersonIcon } from '@*company-data-covered*/design-system';
import { render, screen } from '@testing-library/react';

import MetricsSection from './MetricsSection';
import { METRICS_SECTION_TEST_IDS } from './TestIds';

const metricsSectionProps = {
  title: 'My performance',
  testIdPrefix: 'test',
  icon: <PersonIcon />,
};

const fakeChildProps = {
  dataTestId: 'fakeChild',
  textContent: 'fakeChild',
};

describe('MetricsSection', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MetricsSection {...metricsSectionProps}>
        <div data-testid={fakeChildProps.dataTestId}>
          {fakeChildProps.textContent}
        </div>
      </MetricsSection>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByTestId(`test-${METRICS_SECTION_TEST_IDS.TITLE}`).textContent
    ).toContain(metricsSectionProps.title);

    expect(screen.getByTestId(fakeChildProps.dataTestId).textContent).toContain(
      fakeChildProps.textContent
    );
  });
});
