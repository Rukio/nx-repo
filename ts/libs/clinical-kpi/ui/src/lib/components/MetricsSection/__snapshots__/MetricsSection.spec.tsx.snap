// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsSection should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 2px solid;
  border-bottom-color: rgba(0, 0, 0, 0.12);
  border-bottom-style: solid;
  padding: 4px 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  color: #1976d2;
  margin-left: 12px;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <h6
        class="MuiTypography-root MuiTypography-h6 emotion-3"
        data-testid="test-metrics-section-title"
      >
        My performance
      </h6>
    </div>
    <div
      data-testid="fakeChild"
    >
      fakeChild
    </div>
  </div>
</DocumentFragment>
`;
